Bottom: dbe25949bbbf2ce84e889234850e484e34abf082
Top:    6aaac58a9388080b78b7d9addc12475288dacc4a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 23:08:11 +0000

WORK IN PROGRESS


---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 2b98a75..f5f225a 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -294,13 +294,16 @@ static char *
 nogui_prompt_for (const char *msg, const char *default_string)
 {
   static char buf[1024];
+  char *s;
+
   if (default_string)
     printf ("%s [%s] : ", msg, default_string);
   else
     printf ("%s : ", msg);
-  fgets (buf, 1024, stdin);
-  if (buf[0] == 0 && default_string)
-    strcpy (buf, default_string);
+
+  s = fgets (buf, 1024, stdin);
+  if (default_string && (s == NULL || buf[0] == 0))
+    strncpy (buf, default_string);
   return buf;
 }
 
@@ -312,13 +315,16 @@ nogui_fileselect (const char *title, const char *descr,
 		  const char *history_tag, int flags)
 {
   static char buf[1024];
+  char *s;
+
   if (default_file)
     printf ("%s [%s] : ", title, default_file);
   else
     printf ("%s : ", title);
-  fgets (buf, 1024, stdin);
-  if (buf[0] == 0 && default_file)
-    strcpy (buf, default_file);
+
+  s = fgets (buf, 1024, stdin);
+  if (default_string && (s == NULL || buf[0] == 0))
+    strncpy (buf, default_string);
   return buf;
 }
